datalad/tests/utils.py

```diff
@@ -469,7 +469,12 @@ def newfunc(*arg, **kw):
         flavors_ = _get_resolved_flavors(flavors)
 
         testrepos_uris = _get_testrepos_uris(regex, flavors_)
-        assert(testrepos_uris)
+        # MIH: I don't think we can assert that there always is a testrepo
+        # for network test when NONETWORK is given, this list will be empty
+        # will change it to skipping the test. Remove on ACK.
+        #assert(testrepos_uris)
```

Comment:
Actually, for "with_testrepos" there's a parameter "flavor". Set to 'local_url' in case of DATALAD_NONETWORK. Then testrepos_uris should not be empty, but deliver local urls (file:///where/ever)